examples: add a pgai example for semantic search

Askir opened the pull request:
Simply codified our quickstart into a python script which you can run with one command. The other exampels mostly seem to follow this "python script" approach so I am using it here as well.

Askir committed:
examples: add a pgai example for semantic search

adolsalamanca commented on examples/pgai-semantic-search/docker-compose.yaml:
```diff
@@ -0,0 +1,23 @@
+    ports:
+      - "11434:11434"
+    volumes:
+      - ./ollama:/root/.ollama
```
nit, missing end of file new line

adolsalamanca commented on examples/pgai-semantic-search/main.py:
```diff
@@ -0,0 +1,158 @@
+    conn.close()
+
+if __name__ == "__main__":
+    main()
```
nit, missing end of file new line

adolsalamanca commented on examples/pgai-semantic-search/requirements.txt:
```diff
@@ -0,0 +1,2 @@
+psycopg2-binary>=2.9.9
+requests>=2.31.0
```
nit, missing end of file new line
(not sure if it's  something standard in `requirements.txt` though)

adolsalamanca commented on README.md:
```diff
@@ -408,6 +408,7 @@ See the [API documentation](./docs/api.md) for all endpoints.
 
 - [MindsDB](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/handlers/ollama_handler/README.md) (Connects Ollama models with nearly 200 data platforms and apps)
 - [chromem-go](https://github.com/philippgille/chromem-go/blob/v0.5.0/embed_ollama.go) with [example](https://github.com/philippgille/chromem-go/tree/v0.5.0/examples/rag-wikipedia-ollama)
+- [PGAI](https://github.com/timescale/pgai)(Automatically synchronize embeddings with pgvector)
```
Should we maybe leave it as pgai or Pgai?
```suggestion
- [pgai](https://github.com/timescale/pgai)(Automatically synchronize embeddings with pgvector)
```

adolsalamanca reviewed (approved):
Looks good, left a couple of nits and a comment.

avthars commented on README.md:
```diff
@@ -408,6 +408,7 @@ See the [API documentation](./docs/api.md) for all endpoints.
 
 - [MindsDB](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/handlers/ollama_handler/README.md) (Connects Ollama models with nearly 200 data platforms and apps)
 - [chromem-go](https://github.com/philippgille/chromem-go/blob/v0.5.0/embed_ollama.go) with [example](https://github.com/philippgille/chromem-go/tree/v0.5.0/examples/rag-wikipedia-ollama)
+- [pgai](https://github.com/timescale/pgai) (Automatically synchronize embeddings with pgvector)
```
We should lead with PostgreSQL vs just pgai.

maybe something like:
[PostgreSQL](link...) (Create and sync embeddings from Ollama models using pgvector and pgai)

avthars replied:
I am trying to see how to fit in pgvector, PostgreSQL and pgai 😅 -- lmk your suggestions

avthars commented on examples/pgai-semantic-search/README.md:
```diff
@@ -0,0 +1,38 @@
+
+[pgai](https://github.com/timescale/pgai) is an open-source postgres extension that simplifies the process of building search, retrieval augmented generation (RAG), and other AI applications.
+
+This example demonstrates how to use pgai's vectorizer with ollama to automatically create embeddings for text and perform semantic search on them.
```
pgai vectorizer, not Pgai's vectorizer

avthars commented on examples/pgai-semantic-search/README.md:
```diff
@@ -0,0 +1,38 @@
+2. Connects to PostgreSQL and:
+   1. Installs the pgai extension
+   2. Creates a sample blog table with data
+   3. Sets up a "pgai vectorizer" to automatically create embeddings for the blog posts
```
just "vectorizer" is fine but not a big deal

avthars reviewed (changes requested):
Left a few naming suggestions to fix before an upstream PR

cevian commented on README.md:
```diff
@@ -408,6 +408,7 @@ See the [API documentation](./docs/api.md) for all endpoints.
 
 - [MindsDB](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/handlers/ollama_handler/README.md) (Connects Ollama models with nearly 200 data platforms and apps)
 - [chromem-go](https://github.com/philippgille/chromem-go/blob/v0.5.0/embed_ollama.go) with [example](https://github.com/philippgille/chromem-go/tree/v0.5.0/examples/rag-wikipedia-ollama)
+- [PostgreSQL extension pgai](https://github.com/timescale/pgai) (Create and search embeddings from Ollama models using pgvector)
```
```suggestion
- [the pgai PostgreSQL extension](https://github.com/timescale/pgai) (Create and search embeddings from Ollama models using pgvector)
```

cevian commented on examples/pgai-semantic-search/README.md:
```diff
@@ -0,0 +1,38 @@
+
+[pgai](https://github.com/timescale/pgai) is an open-source postgres extension that simplifies the process of building search, retrieval augmented generation (RAG), and other AI applications.
+
+This example demonstrates how to use pgai vectorizer with ollama to automatically create embeddings for text and perform semantic search on them.
```
```suggestion
This example demonstrates how to use [pgai vectorizer](https://www.timescale.com/blog/vector-databases-are-the-wrong-abstraction/) with ollama to automatically create embeddings for text and perform semantic search on them.
```

cevian reviewed (approved):
Minor suggestions

Askir committed:
docs: switch around database integrations order and link to quickstart

Askir committed:
docs: link to blog post in example readme

Askir committed:
chore: link to main readme

mchiang0610 committed:
readme: removing example to link externally

readme: removing example to link externally so we don't have to keep this example up-to-date
